xrpl4j-address-codec/src/test/java/org/xrpl/xrpl4j/codec/addresses/UnsignedByteArrayTest.java

```diff
@@ -41,4 +41,23 @@ public void lowerCaseOrUpperCase() {
     assertThat(UnsignedByteArray.fromHex("abcdef0123").hexValue()).isEqualTo("ABCDEF0123");
   }
 
+  @Test
+  public void unsignedByteArrayEqualsTest() {
+    UnsignedByteArray array1 = UnsignedByteArray.of(new byte[] {0, MAX_BYTE});
+    UnsignedByteArray array2 = UnsignedByteArray.of(new byte[] {MAX_BYTE, 0});
+    UnsignedByteArray array3 = UnsignedByteArray.of(new byte[] {0, MAX_BYTE});
+    UnsignedByteArray array4 = array1;
+
+    assertThat(array1.equals(array1)).isTrue();
+    assertThat(array1.equals(array2)).isFalse();
+    assertThat(array1.equals(array3)).isTrue();
+    assertThat(array1 == array3).isFalse();
+    assertThat(array1.toByteArray() == array3.toByteArray()).isFalse();
```

Comment:
Minor nit, but let's throw in the following just be sure `toByteArray` is getting executed properly:

```
assertThat(Arrays.equals(array1.toByteArray(), array3.toByteArray())).isTrue();
```